appium/webdriver/webdriver.py

```diff
@@ -581,8 +582,23 @@ def lock(self, seconds):
             'seconds': seconds,
         }
         self.execute(Command.LOCK, data)
+        if self.desired_capabilities.get('platformName', '').lower() == 'android':
+            WebDriverWait(self, 10).until(lambda d: d.is_locked())
```

Comment:
so for now, if it's flakey, we can just show users how to add the wait in their own client code. meanwhile, we'll work on making the appium server more responsible :-)